sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosBatchRequestOptions.java

```diff
@@ -206,7 +207,9 @@ Map<String, String> getHeaders() {
      * @return the current request options.
      */
     public CosmosBatchRequestOptions setKeywordIdentifiers(Set<String> keywordIdentifiers) {
-        this.keywordIdentifiers = keywordIdentifiers;
+        if (keywordIdentifiers != null) {
```

Comment:
what if keywordIdentifiers is null - should we treat it as signal to clear all values or instead throw a NullPointerException?

Reply:
Changed to creating a new set as clearing it would not work as it is immutable.